bot/cogs/help_channels.py

```diff
@@ -754,9 +737,24 @@ async def on_message_delete(self, msg: discord.Message) -> None:
         self.scheduler.schedule_later(delay, msg.channel.id, self.move_idle_channel(msg.channel))
 
     async def is_empty(self, channel: discord.TextChannel) -> bool:
-        """Return True if the most recent message in `channel` is the bot's `AVAILABLE_MSG`."""
-        msg = await self.get_last_message(channel)
-        return self.match_bot_embed(msg, AVAILABLE_MSG)
+        """Return True if there's an AVAILABLE_MSG and the messages leading up are bot messages."""
+        log.trace(f"Checking if #{channel} ({channel.id}) is empty.")
+        found = False
+
+        # A limit of 100 results in a single API call.
+        # If AVAILABLE_MSG isn't found within 100 messages, then assume the channel is not empty.
+        # Not gonna do an extensive search for it cause it's too expensive.
+        async for msg in channel.history(limit=100):
+            if not msg.author.bot:
+                log.trace(f"#{channel} ({channel.id}) has a non-bot message.")
+                return False
+
+            if self.match_bot_embed(msg, AVAILABLE_MSG):
+                log.trace(f"#{channel} ({channel.id}) has the available message embed.")
+                found = True
+                break
+
+        return found
```

Comment:
Usually when I see this pattern I try to refactor it into just returning directly, i.e. instead of:
```python
found = False

for x in y:
  if z:
    found = True
    break

return found
```
I think it's nicer to just:
```python
for x in y:
  if z:
    return True

return False
```

Reply:
There was a reason for this before but I don't remember anymore. Probably a previous version of the code.